src/PrMerger.js

```diff
@@ -93,51 +90,25 @@ class PrMerger {
         Logger.info("PR processing order:", this._prNumbers());
     }
 
-    // remembers still-relevant PR tags and
-    // deletes (from GitHub) PR tags which do not have a corresponding open PR
-    async _importTags(rawTags) {
-        assert(rawTags);
-
-        assert(!this._tags);
-        this._tags = [];
-
-        for (let tag of rawTags) {
-            let prNum = Util.ParseTag(tag.ref);
-            if (prNum === null)
-                continue;
-            for (let pr of this._todo) {
-                if (prNum === pr.number.toString()) {
-                    prNum = null;
-                    this._tags.push(tag);
-                    break;
-                }
-            }
-            if (prNum !== null) {
-                if (!Config.dryRun())
-                    await GH.deleteReference(Util.StagingTag(prNum));
-            }
-        }
-    }
-
     // Returns a raw PR with a staged commit (or null).
     // If that PR exists, it is in either a "staged" or "merged" state.
-    // Requires _importTags() being run first.
     async _current() {
         Logger.info("Looking for the current PR...");
-        const stagingSha = await GH.getReference(Config.stagingBranchPath());
-        // search for a tag, the staging_branch points to,
-        // and parse out PR number from the tag name
-        const tag = this._tags.find((t) => { return (t.object.sha === stagingSha) && Util.MatchTag(t.ref); });
-        if (tag === undefined) {
-            Logger.info("No current PR found.");
-            return null;
+        const stagedBranchSha = await GH.getReference(Config.stagingBranchPath());
+        const stagedBranchCommit = await GH.getCommit(stagedBranchSha);
+        Logger.info("Staged branch head sha: " + stagedBranchCommit.sha);
+        const prNum = Util.ParsePrNumber(stagedBranchCommit.message);
+        if (prNum === null) {
```

Comment:
The checks for ParsePrNumber() results are inconsistent: Some callers compare with `null` and some calculate an implicit thruthy-ness of the value. I think it is better _not_ to compare with null explicitly because it will cover the (currently impossible?) case where prNum is undefined. I would also assert that a found prNum is always positive inside ParsePrNumber().

Reply:
I reworked this code after addressing another request.

>  I would also assert that a found prNum is always positive inside ParsePrNumber().

Done.